compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -2994,7 +2994,7 @@ class Typer extends Namer
         SearchFailure(qual.withType(NestedFailure(ex.toMessage, selectionProto))))
 
     // try an implicit conversion or given extension
-    if ctx.mode.is(Mode.ImplicitsEnabled) && qual.tpe.isValueType then
+    if ctx.mode.is(Mode.ImplicitsEnabled) && tree.name != nme.CONSTRUCTOR && qual.tpe.isValueType then
```

Comment:
```suggestion
    if ctx.mode.is(Mode.ImplicitsEnabled) && !tree.name.isConstructorName && qual.tpe.isValueType then
```
(this also includes trait constructors, I don't know if it matters here but using the method is cleaner anyway)